### ISSUE TITLE
OCI HDFS connector in read-ahead mode doesn't emit `bytesRead` input metric 

### ISSUE THREAD

**Issue by kvirund** [oracle/oci-hdfs-connector#77]
OCI HDFS connector in read-ahead mode doesn't emit `bytesRead` input metric:

We were using default mode and were reading the number of read rows and read bytes using the following metrics:
* `stageCompleted.stageInfo.taskMetrics.inputMetrics.bytesRead`
* `stageCompleted.stageInfo.taskMetrics.inputMetrics.recordsRead`

inside a listener's event `override def onStageCompleted(stageCompleted: SparkListenerStageCompleted)`.

But after enabling "read ahead" mode, the `bytesRead` metric is always zero.

**Comment by kvirund** [comment]
Implementations should be changed to:

```
    @Override
    public int read() throws IOException {
        LOG.debug("{}: Reading single byte at position {}", this, filePos);
        if (dataPos == -1) {
            fillBuffer();
        }
        if (dataPos == -1) {
            return -1;
        }
        filePos++;
        int result = Byte.toUnsignedInt(data[dataCurOffset++]);
        this.statistics.incrementBytesRead(1);
        if (atEndOfBuffer()) {
            clearBuffer();
        }
        return result;
    }
```

and

```
    @Override
    public int read(byte[] buffer, int offset, int length) throws IOException {
        LOG.debug("{}: Reading {} bytes at current position {}", this, length, filePos);
        if (dataPos == -1) {
            fillBuffer();
        }
        if (dataPos == -1) {
            return -1; // EOF
        }
        int n = Math.min(length, dataMax - dataCurOffset);
        System.arraycopy(data, dataCurOffset, buffer, offset, n);
        dataCurOffset += n;
        filePos += n;
        if (atEndOfBuffer()) {
            if (n != length) {
                LOG.debug("{}: Short Read; exhausted buffer", this);
            }
            clearBuffer();
        }
        this.statistics.incrementBytesRead(n);
        return n;
    }
```

**Comment by kvirund** [comment]
Tested it by referencing patched implementation in `fs.oci.client.apache.connection.closing.strategy`.

**Comment by anton-gorev-oracle** [comment]
PR: #78

**Comment by mricken** [comment]
This was released in version [3.3.4.0.0.0](https://github.com/oracle/oci-hdfs-connector/releases/tag/v3.3.4.0.0.0). Closing.
